### PR TITLE
Add user guide for the unit testing framework

### PR DESCRIPTION
**Pull request by sajithaliyanage**
## Purpose
Add a user guide for the synapse unit testing framework including details of how to start and stop framework and sample test cases.

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: gh-docs/synapse-unit-testing-framework/synapse-unit-testing-framework.md [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+
+By default the unit testing framework starts on port 9007 and you can change the starting port by passing the port as above
+
+### Stopping Unit Testing Framework
+To stop the Unit Testing Framework runtime, press Ctrl+C in the command window and it will stop the server along with the Enterprise Integrator runtime.
+
+### How To Send a TCP Request
+
+You can use following code segment to send a sample test case to the unit testing framework and get the server response.
+```
```
Comment (menuka94) [inline gh-docs/synapse-unit-testing-framework/synapse-unit-testing-framework.md]: Shall we use language-specific syntax highlighting here? i.e. 
```
```java
```

File: gh-docs/synapse-unit-testing-framework/synapse-unit-testing-framework.md [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+} 
+```
+
+## Sample Test Cases
+Each request needs to be sent to the unit testing server via a TCP transport (user defined port or default 9007 port). 
+
+### Sample Test Case for Sequence Artifact
+
+Client request via TCP:
+```
```
Comment (menuka94) [inline gh-docs/synapse-unit-testing-framework/synapse-unit-testing-framework.md]:
```suggestion
```xml
```

File: gh-docs/synapse-unit-testing-framework/synapse-unit-testing-framework.md [hunk 3]
```diff
@@ -0,0 +1,375 @@
...
+
+Response from unit testing server:
+```
+{'test-cases':'SUCCESS'}
+```
+
+### Sample Test Case for Proxy Artifact
+
+Client request via TCP:
+```
```
Comment (menuka94) [inline gh-docs/synapse-unit-testing-framework/synapse-unit-testing-framework.md]:
```suggestion
```xml
```

File: gh-docs/synapse-unit-testing-framework/synapse-unit-testing-framework.md [hunk 4]
```diff
@@ -0,0 +1,375 @@
...
+</synapse-unit-test>
+```
+Response from unit testing server:
+```
+{'test-cases':'SUCCESS'}
+```
+### Sample Test Case for API Artifact
+
+Client request via TCP:
+```
```
Comment (menuka94) [inline gh-docs/synapse-unit-testing-framework/synapse-unit-testing-framework.md]:
```suggestion
```xml
```

File: gh-docs/synapse-unit-testing-framework/synapse-unit-testing-framework.md [hunk 5]
```diff
@@ -0,0 +1,375 @@
...
+If a unit test failed while reading input data, artifact deployment, mock-service creation, test case mediation or assertion, framework responses the error message to the client as follows.
+
+```
+{"<failed-phase>":"failed","exception":"<custom assertion or java exception message>"}
+```
+
+Following is an example of a custom error message sent by the unit testing framework due to un-matching actual payload and expected payload.
+
+Client request via TCP:
+```
```
Comment (menuka94) [inline gh-docs/synapse-unit-testing-framework/synapse-unit-testing-framework.md]:
```suggestion
```xml
```

File: gh-docs/synapse-unit-testing-framework/synapse-unit-testing-framework.md (hunk 1 above)
Comment (sajithaliyanage) [reply gh-docs/synapse-unit-testing-framework/synapse-unit-testing-framework.md]: Added

File: gh-docs/synapse-unit-testing-framework/synapse-unit-testing-framework.md (hunk 2 above)
Comment (sajithaliyanage) [reply gh-docs/synapse-unit-testing-framework/synapse-unit-testing-framework.md]: Added

File: gh-docs/synapse-unit-testing-framework/synapse-unit-testing-framework.md (hunk 3 above)
Comment (sajithaliyanage) [reply gh-docs/synapse-unit-testing-framework/synapse-unit-testing-framework.md]: Added

File: gh-docs/synapse-unit-testing-framework/synapse-unit-testing-framework.md (hunk 5 above)
Comment (sajithaliyanage) [reply gh-docs/synapse-unit-testing-framework/synapse-unit-testing-framework.md]: Added

File: gh-docs/synapse-unit-testing-framework/synapse-unit-testing-framework.md (hunk 4 above)
Comment (sajithaliyanage) [reply gh-docs/synapse-unit-testing-framework/synapse-unit-testing-framework.md]: Added

File: gh-docs/synapse-unit-testing-framework/synapse-unit-testing-framework.md [hunk 6]
```diff
@@ -0,0 +1,371 @@
+# Synapse Unit Testing Framework
+Unit testing framework for synapse provides an in-built way to execute a unit test against a given synapse configuration artifacts such as sequence, proxy, api, local entry, endpoint. Mainly it required a test artifact which is your main configuration, supportive artifacts which needs to execute the test artifact if exists and test cases with assertions. 
+
+### Features
```
Comment (Nirothipan) [inline gh-docs/synapse-unit-testing-framework/synapse-unit-testing-framework.md]: s
```suggestion
### Features
```

File: README.md [hunk 7]
```diff
@@ -156,6 +156,9 @@ work on developing integration flows.
 
 Read on how to develop integration artifacts with EI tooling [here](https://docs.wso2.com/display/EI640/Integration+Tutorials).
 
+#### Test Artifacts in the Enterprise Integrator
+WSO2 Enterprise Integrator allows you to execute unit tests for your synapse configurations by using [Synapse Unit Testing Framework](gh-docs/synapse-unit-testing-framework/synapse-unit-testing-framework.md).
```
Comment (abeykoon) [inline README.md]: Shall we update to have "unit tests against your integration artifacts by using"

Comment (sajithaliyanage) [reply README.md]: Edited

**Pull request merged**